syncclient/client.py

```diff
@@ -43,18 +43,65 @@ def get_browserid_assertion(login, password, fxa_server_url=FXA_SERVER_URL,
     return bid_assertion, hexlify(sha256(keyB.encode('utf-8')).digest()[0:16])
 
 
+class SyncClientError(Exception):
+    """An error occured in SyncClient."""
+
+
+class TokenserverClient(object):
+    """Client for the Firefox Sync Token Server.
+    """
+    def __init__(self, bid_assertion, client_state,
+                 server_url=TOKENSERVER_URL):
+        self.bid_assertion = bid_assertion
+        self.client_state = client_state
+        self.server_url = server_url
+
+    def get_hawk_credentials(self):
+        """Asks for new temporary token given a BrowserID assertion"""
+        authorization = 'BrowserID %s' % encode_header(self.bid_assertion)
+        headers = {
+            'Authorization': authorization,
+            'X-Client-State': self.client_state
+        }
+        url = urlparse.urljoin(self.server_url, '/1.0/sync/1.5')
+        raw_resp = requests.get(url, headers=headers)
+        raw_resp.raise_for_status()
+        return raw_resp.json()
+
+
 class SyncClient(object):
     """Client for the Firefox Sync server.
     """
 
-    def __init__(self, bid_assertion, client_state,
-                 tokenserver_url=TOKENSERVER_URL,
-                 fxa_server_url=FXA_SERVER_URL):
-        self._authenticate(bid_assertion, client_state, tokenserver_url)
+    def __init__(self, bid_assertion=None, client_state=None,
+                 tokenserver_url=TOKENSERVER_URL, **credentials):
+
+        credentials_complete = set(credentials.keys()).issuperset({
```

Comment:
If a key is missing it means the developer misused the lib and we fail as soon as we recognize it.